### ISSUE TITLE
Unsupported coverage table format 100

### ISSUE THREAD

**Issue by Omikhleia** [sile-typesetter/sile#1222]
I have a simple text document with at its beginning, a global:

`\font[family=Linux Libertine O]`

SILE 0.11.1 = works fine, and the document _seems_ to be ok.

Current master (as of 0cb304f7eba46b81a1e46914ccc7603ea18e08cb) = aborts on error:

`! Unsupported coverage table format 100 at test.sil:4:1: in \fonttable: 0x560e3a4016e0`

Something might have been broken on master - maybe it should just be a warning and/or have some more clear message?

(FWIW, I'm switching my documents to Libertinus - which doesn't have this issue - but just reporting an issue in case some other fonts people might be tempted to use just stop working. I don't really care about "Linux Libertine O", but other fonts might fail too with this obscure message, I guess, and it's always good to know about breaking changes if it is not addressed.

**Comment by ctrlcctrlv** [comment]
Are you sure that this isn't expected behavior? The font sounds broken—there are only two valid values in the OpenType spec for coverage table format, 1 and 2. https://docs.microsoft.com/en-us/typography/opentype/spec/chapter2#coverage-table

**Comment by simoncozens** [comment]
Either the font is broken or (more likely) SILE is having a hard time parsing it.

**Comment by simoncozens** [comment]
Please turn on debugging for the opentype parser (`sile --debug=opentype-parser`) and post the full log.

**Comment by Omikhleia** [comment]
@ctrlcctrlv 
>Are you sure that this isn't expected behavior? The font sounds broken—there are only two valid values in the OpenType spec for coverage table format, 1 and 2.

I have no idea. Could be broken (using a non-standard table) or could be something else.

These fonts are just part of the standard Ubuntu package https://packages.ubuntu.com/search?keywords=libertine

@simoncozens 
> Please turn on debugging for the opentype parser (`sile --debug=opentype-parser`) and post the full log.

```
[opentype-parser] header = {majorVersion=1,mathConstantsOffset=10,mathGlyphInfoOffset=224,mathVariantsOffset=232,minorVersion=0}
[opentype-parser] mathGlyphInfoOffset = 224
[opentype-parser] mathGlyphInfo = {extendedShapeCoverageOffset=0,mathItalicsCorrectionInfoOffset=0,mathKernInfoOffset=0,mathTopAccentAttachmentOffset=0}
! Unsupported coverage table format 100 at frtest.sil:5:1: in \fonttable: 0x55586dc325d0
```
Switching the error into a warning doesn't help

```
(...same as above...)
Error detected:
	core/opentype-parser.lua:364: attempt to get length of local 'horizGlyphCoverage' (a nil value)
```

**Comment by simoncozens** [comment]
Probably a bug in the MATH table reader. I’ll take a look.

**Comment by alerque** [comment]
I'm pretty sure the *Linux Libertine* family had a botched partial MATH implementation. That was part of the initial impetus to fork it to Libertinus (Libertinus Math now being SILE's default math font).

That being said this is a regression since the last release that we should probably fix before releasing math. Now that we're parsing the MATH table instead of just ignoring it, we shouldn't be crashing when using fonts with borked math properties unless we're actually using them for math.

**Comment by simoncozens** [comment]
The problem is that in Linux Libertine the `MathVariants.horizGlyphCoverageOffset` is NULL. The spec doesn't *say* that it can be NULL. But clearly when `MathVariants.horizGlyphCount` is 0, the coverage table is going to be empty. I've fixed this in #1230, and I'll raise a question with OpenType whether the font is conformant.
